packages/backend/src/services/OrganizationService/OrganizationService.ts

```diff
@@ -96,6 +102,43 @@ export class OrganizationService {
         });
     }
 
+    async delete(organizationUuid: string, user: SessionUser): Promise<void> {
+        const organization = await this.organizationModel.get(organizationUuid);
+        if (
+            user.ability.cannot(
+                'delete',
+                subject('Organization', { organizationUuid }),
+            )
+        ) {
+            throw new ForbiddenError();
+        }
+
+        const orgUsers =
+            await this.organizationMemberProfileModel.getOrganizationMembers(
+                organizationUuid,
+            );
+
+        const deleteUserPromises = orgUsers.map((orgUser) =>
+            this.userModel.delete(orgUser.userUuid),
+        );
+        await Promise.all(deleteUserPromises);
+
+        await this.organizationModel.delete(organizationUuid);
```

Comment:
do we need to delete anything else? or associations will take care of it?

Reply:
Most things are deleted on cascade. Except users, because they are separate entities.

Reply:
nice!

Reply:
Perhaps we should delete the org and users in the same database transaction. 
With the current code, if the org fails to delete we already deleted the users.

Reply:
Yep, good idea, I think I can move the `delete user` to the org modal and rename  `deleteOrgAndUsers`, what do you think ?

Reply:
sounds perfect 👍

Reply:
Done